test/e2e/prometheus_test.go

```diff
@@ -64,6 +66,31 @@ func TestPrometheusMetrics(t *testing.T) {
 	}
 }
 
+// TestPrometheusGOGC verifies that Prometheus containers use the appropriate gogc value. It also helps us stay informed
+// about any changes to the default gogc value upstream. If that value changes, this test will fail, prompting us to
+// consider whether the new default value is still suitable.
+// Refer to this link for some points that may need to be examined https://github.com/openshift/prometheus/pull/206#issuecomment-2182168575.
+func TestPrometheusGOGC(t *testing.T) {
+	infra, err := f.OperatorClient.GetInfrastructure(ctx, "cluster")
+	require.NoError(t, err)
+	infraConfig := operator.NewInfrastructureConfig(infra)
+
+	gogc := 100
+	if infraConfig.HighlyAvailableInfrastructure() {
```

Comment:
note that CMO e2e tests only run in HA mode (not sure if they would even pass with SNO).

Reply:
I had a doubt about that, thanks, that will make the test simpler.